### DISCUSSION TITLE
What is the file structure for WHAM to run the script? ----as well as a minor feature request on file structure for data loading

### DISCUSSION THREAD

**Discussion by underdogliu** [speechbrain/speechbrain#1662] [Q&A]
I am now trying to train the model using WHAM dataset for speech enhancement. However, while the wavs and metadata are there, I found my file structure is not applicable for the script to run. Therefore, I am wondering if we have a sample file structure (using `tree` to demo will be perfectly fine) is available for reference.

Also, this is not only applied to WHAM in particular, but also for other datasets - lacking a description on file structure from the original dataset. SO I wonder if we can have or already have such thing in-place. If not I suggest to write some instructions about this.

Thanks in advance!

**Comment by ycemsubakan** [comment]
Hi @underdogliu , sorry for my late reply, and thanks for your suggestion. It would indeed be great to have this in the README of each recipe. I will try to get to this. 

In the meanwhile, basically what you need to do is simply modify the csv file generation script that that generates the csv file that contains the paths of the files for the dataloader. 

You can check this: 
https://github.com/speechbrain/speechbrain/blob/a9b373a9fef3fdfc4e3b4a41478f7b283866186f/recipes/WSJ0Mix/prepare_data.py#L55

Basically you can hack this function to accommodate your needs.

**Comment by poor1017** [comment]
.
└── whamr
    └── wav16k
        ├── max
        └── min
            ├── cv
            │   ├── mix_both_anechoic
            │   ├── mix_both_reverb
            │   ├── mix_clean_anechoic
            │   ├── mix_clean_reverb
            │   ├── mix_single_anechoic
            │   ├── mix_single_reverb
            │   ├── noise
            │   ├── s1_anechoic
            │   ├── s1_reverb
            │   ├── s2_anechoic
            │   └── s2_reverb
            ├── tr
            │   ├── mix_both_anechoic
            │   ├── mix_both_reverb
            │   ├── mix_clean_anechoic
            │   ├── mix_clean_reverb
            │   ├── mix_single_anechoic
            │   ├── mix_single_reverb
            │   ├── noise
            │   ├── s1_anechoic
            │   ├── s1_reverb
            │   ├── s2_anechoic
            │   └── s2_reverb
            └── tt
                ├── mix_both_anechoic
                ├── mix_both_reverb
                ├── mix_clean_anechoic
                ├── mix_clean_reverb
                ├── mix_single_anechoic
                ├── mix_single_reverb
                ├── noise
                ├── s1_anechoic
                ├── s1_reverb
                ├── s2_anechoic
                └── s2_reverb
